File: backend/functools/path_from_id_generator.go
package functools

import (
	"strings"
)

func PathFromIdGenerator(id string) string {
	sb := strings.Builder{}
	idLen := len(id)
	twoLetterSize := idLen >> 1
	for i := 0; i < twoLetterSize; i++ {
		sb.WriteByte('/')
		sb.WriteByte(id[idLen-2*i-1])
		sb.WriteByte('/')
		sb.WriteByte(id[idLen-2*i-2])
	}

	if idLen != twoLetterSize<<1 {
		sb.WriteByte('/')
		sb.WriteByte(id[0])
	}
	return sb.String()
}

File: [source] frontend/functools/PathFromIdGenerator.js
export default function PathFromIdGenerator(id) {
    let result = ""
    try{
        id = id.toString()
        const idLen = id.length
        const twoLetterSize = idLen >> 1

        for (let i= 0; i < twoLetterSize; i++){
            result += `/${id[idLen - 2*i - 1]}/${id[idLen - 2*i - 2]}`
        }

        if (idLen !== twoLetterSize << 1){
            result += `/${id[0]}`
        }
    }catch (e) {
        console.error("ERROR FROM: [PathFromIdGenerator]")
        return null
    }

    return result
}